enhancements/helm3/console.md

```diff
@@ -0,0 +1,239 @@
+
+In the initial phase, the chart repository would be served of [redhat-helm-charts](https://redhat-developer.github.io/redhat-helm-charts) public [GitHub repository](https://github.com/redhat-developer/redhat-helm-charts).
+
+New charts will be added and/or existing curated by submitting PRs against the above mentioned GitHub repository.
+
+### How would the UI discover the charts
+
+1. The UI would invoke `/api/helm/charts/index.yaml` endpoint to get [the repository index file](https://helm.sh/docs/topics/chart_repository/#the-index-file) so that the available charts can be rendered in the developer catalog. 
+
+2. The above endpoint would proxy requests to the configured chart repository
+
+
+![Helm Charts Repo Service](../helm3/assets/charts-repo.png)
+
+### How would disconnected installs work 
+
+1. The user would need to 'clone' the content of the chart repository over the fence
+
+   * The public GitHub repository could be cloned into inside-the-network GitHub or Gitlab instance and configured to serve static content ( "Pages" ).
+   * The content of the chart repository could be crawled and served using a (containerized or external) HTTP server, e.g. nginx
+
+2. The URL serving the above static content would need to be passed to chart repository proxy running inside the cluster. 
+
+### Configuring Helm Chart Repository location
+
+Since the experience we are building needs a representation of the chart repository in-cluster, there needs to be a standard way to define a Kubernetes resource for the same.
+
+Configuring Helm repository location could be modeled similar to [`OperatorSource`](https://github.com/operator-framework/operator-marketplace/blob/7d230952a1045624b7601b4d6e1d45b3def4cf76/deploy/crds/operators_v1_operatorsource_crd.yaml). 
+Due to future planned federated usecases ([ODC-2994](https://issues.redhat.com/browse/ODC-2994)), a cluster admin should be able to declare multiple chart repositories.
+
+#### Existing/Known In-Cluster Representations of Helm Chart Repositories
+
+[Open Cluster Management project](https://github.com/open-cluster-management/multicloud-operators-subscription) introduced a notion of subscription for [a Helm chart](https://github.com/open-cluster-management/multicloud-operators-subscription/tree/master/examples/helmrepo-channel). A subscription is defined on the top of a channel of [type `HelmRepo`](https://github.com/open-cluster-management/multicloud-operators-subscription/blob/master/examples/helmrepo-channel/01-channel.yaml). Currently, chart repo configuration is provided in a separate `ConfigMap`, but the model could be extended to use the configuration provided in referred `HelmChartRepository` instance.
+
+#### Introducing the HelmChartRepository API
+
+The `HelmChartRepository` is a simple cluster-scoped API for letting users/admins define Helm Chart Repository configurations in a cluster. 
+
+```yaml
+apiVersion: helm.openshift.io/v1alpha1
```

Comment:
Sorry if this was discussed above and I missed it, but is this meant to be an alpha API? What is the intended support level?

Reply:
There was discussion in the API PR and we went for Beta
https://github.com/openshift/api/pull/598/files#diff-ed4a5a3bba15a3929122e018a72df324R15

Good catch, this part needs an update.

Reply:
Updated.